### ISSUE TITLE
Incorrect find_replace_templatesets() functionality

### ISSUE THREAD

**Issue by Destroy666x** [mybb/mybb#1903]
The following code is incorrect and mixed: https://github.com/mybb/mybb/blob/feature/inc/adminfunctions_templates.php#L33-55 Only the 2 first lines are correct inside the conditional.

And, in my opinion, global templates should always be modified by the function (unless a specific `$sid` is chosen). So this should consider `-1` too: https://github.com/mybb/mybb/blob/feature/inc/adminfunctions_templates.php#L29

More info:
http://community.mybb.com/thread-168112.html
http://community.mybb.com/thread-168051.html

**Comment by martec** [comment]
@Destroy666x 
i not know if i making something wrong...
but for me this not work correctly...
same code when i try add in common template add correctly....
but when i try add in global template, mybb 1.8.5 add same thing thrice.
Please could you test this again?

**Comment by Sama34** [comment]
Wasn't this already pushed in 1.8.5 release? You should open a new issue if you can replicate.

**Comment by Stefan-ST** [comment]
@martec Is your problem related to #2016?

**Comment by martec** [comment]
@Stefan-ST 
in 1.8.4 or before not add anything in global template...
now in 1.8.5 after this fix #2016, add same thing thrice.
In common template work correctly. But global template not work correctly.

**Comment by Destroy666x** [comment]
@martec, how do you call the function so that it adds something 3 times?

**Comment by martec** [comment]
@Destroy666x 
something below:

```
		find_replace_templatesets(
			'codebutquick',
			'#' . preg_quote('maximize,') . '#i',
			'"+newbutbar+"maximize,'
		);
```

If i made samething with common template add only once, but if i made with global template add 3 times samething. In example above add +newbutbar+ 3 times.

**Comment by martec** [comment]
To make all thing more easy plugin is https://github.com/martec/myinsertbuttons/blob/master/inc/plugins/myinsertbuttons.php

in 1.8.4 i said in thread of plugin http://community.mybb.com/thread-168909.html to users change $sqlwhere = '>\'0\'';  to $sqlwhere = '>\'-2\''; . If made this in 1.8.4, add only once (so correctly). But in 1.8.5 add 3 times same thing.

**Comment by Destroy666x** [comment]
I added this to global templates: https://github.com/martec/quickadveditor/blob/master/inc/plugins/quickadveditor.php#L188373 and run this once:
```php
        find_replace_templatesets(
            'codebutquick',
            '#' . preg_quote('maximize,') . '#i',
            '"+newbutbar+"maximize,'
        );
```
The result is fine:
`	toolbar: "{$basic1}{$align}{$font}{$size}{$color}{$removeformat}{$basic2}image,{$email}{$link}|video{$emoticon}|{$list}{$code}quote|"+newbutbar+"maximize,source",`

**Comment by martec** [comment]
@Destroy666x 
thanks... i not know about fix of 1.8.6...
i will test... 
so real fix in 1.8.6?

**Comment by Destroy666x** [comment]
Yes, someone linked a PR above

**Comment by martec** [comment]
@Destroy666x 
sorry i not see correctly...
and thanks.
